src/Lykke.Service.IroncladDecorator/Sessions/UserSession/UserSession.cs

```diff
@@ -1,47 +1,30 @@
 ﻿using System;
-using System.Collections.Generic;
-using System.Collections.ObjectModel;
-using Newtonsoft.Json;
+using Lykke.Service.Session.Client;
+using MessagePack;
 
 namespace Lykke.Service.IroncladDecorator.Sessions
 {
+    [MessagePackObject(true)]
     public class UserSession
     {
-        private readonly IDictionary<string, string> _data;
-
-        public UserSession(string id, IDictionary<string, string> data)
-        {
-            Id = id;
-            _data = data;
-        }
+        public string Id { get; set; }
+        public string OldLykkeToken { get; set; }
+        public string LykkeClientId { get; set; }
+        public Guid AuthId { get; set; }
+        public TokenData IroncladTokenResponse { get; set; }
+        public string AuthorizeQuery { get; set; }
 
         public UserSession()
         {
             Id = Guid.NewGuid().ToString("N");
-            _data = new Dictionary<string, string>();
-        }
-
-        public string Id { get; }
-
-        public IReadOnlyDictionary<string, string> Data => new ReadOnlyDictionary<string, string>(_data);
-
-        public void Set<T>(string key, T value)
-        {
-            _data[key] = JsonConvert.SerializeObject(value);
-        }
-
-        public T Get<T>(string key)
-        {
-            _data.TryGetValue(key, out var value);
-
-            var deserialized = JsonConvert.DeserializeObject<T>(value);
-
-            return deserialized;
         }
 
-        public bool Remove(string key)
+        public void SaveAuthResult(IClientSession clientSession, TokenData tokens)
         {
-            return _data.Remove(key);
+            OldLykkeToken = clientSession.SessionToken;
```

Comment:
I think the `LykkeSession` class could be used to store `clientSession` response.
There is no need for storing `IroncladTokenResponse` inside of UserSession. We store only `OldLykkeToken` in UserSession by which we could get associated ironclad tokens.

Reply:
And authId will remain here as well